src/Units/UnitModelScheme.java

```diff
@@ -0,0 +1,62 @@
+package Units;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import Terrain.TerrainType;
+import Units.UnitModel.UnitRoleEnum;
+
+/**
+ * Provides the primary interface for the game to get info about game-specific units.
+ */
+public abstract class UnitModelScheme implements Serializable
+{
+  private static final long serialVersionUID = 1L;
+  
+  /** Indicates whether, for the given map/game, all the units can be instantiated */
+  public boolean schemeValid = true;
+
+  /**
+   * When the UI wants to show a unit type, it should use this string.
+   * Typically the name of whatever passes for infantry, but can be anything.
+   */
+  public abstract String getIconicUnitName();
+  
+  // Fetches a shopping list and a full unit list for a generic commander.
+  public abstract GameReadyModels getGameReadyModels();
+
+  // Holds the data for a single Commander's unit selection.
+  public static class GameReadyModels
+  {
+    public HashMap<TerrainType, ArrayList<UnitModel>> shoppingList = new HashMap<TerrainType, ArrayList<UnitModel>>();
+    public ArrayList<UnitModel> unitModels = new ArrayList<UnitModel>();
+  }
+  
+  public static UnitModel getModelFromString(String pName, ArrayList<UnitModel> models)
+  {
+    String name = UnitModel.standardizeID(pName);
+    UnitModel model = null;
+    ArrayList<String> coreRoles = UnitModel.getUnitRoleIDs();
+    if (coreRoles.contains(name))
```

Comment:
Instead of calling `contains` and `indexOf`, I'd probably just call `indexOf` and compare the return to `-1` to avoid iterating the collection twice. On the other hand, this isn't performance-critical code, and this is arguably more idiomatic.